Agenda for 2024-09-04 CG meeting (#680)
solid/specification: meetings/2024-09-04.md

# W3C Solid Community Group: Weekly

* Date: 2024-09-04T14:00:00Z
* Call: https://meet.jit.si/solid-cg
* Chat: https://matrix.to/#/#solid_specification:gitter.im
* Repository: https://github.com/solid/specification

> * Status: Draft

Reviewer: ```suggestion
* Status: Published
```

* SC: Some of the key spec received updates, even Solid Protocol had a newer version. I can agree with you but about the implementation experience. There were a lot of calls about implementation experience. Considering the number of the active participants to the number of implementers doesn't seem to add up. There is difference what is visible to the CG and what is visible to the outside. Some organizations may be claiming implementing the Solid Protocol but without providing feedback to the CG. I can agree that the community group slowed down but I think do the lack of the implementation feedback. I'm curious how the implementation experience will come up in the WG. Besides that we had a number of updates to the specifications.
* HZ: I agree with you, there may be other factors like 
* eP: There is no objective way to measure it, in my subj
* PAC: I don't have an answer, when Sarven asks if the implementation experience will increase. From my experience some organizations are cautious with technologies which are not on the REC track. 
* SC: This is a very likely scenario. As far as the communication, there are things being implemented but it isn't communicated in the CG.

> * HZ: This topic was mostly about the announcemtent and celebrating it.
> * HZ: As for implementations we have Manas and ActivityPods which were very active in last year. I think there are implemenations on the solid server side.
> * eP: We have feedback, for example https://forum.solidproject.org/t/solid-interop-in-practice/7701

Reviewer 2: ```suggestion
* HZ: This topic was mostly about the announcement and celebrating it.
* HZ: As for implementations, we have Manas and ActivityPods which were very active in last year. I think there are also implementations on the Solid server side.
* eP: We have feedback; for example, https://forum.solidproject.org/t/solid-interop-in-practice/7701
```

* TBL: ... 

### TPAC Demos

* VB (not present, via HackMD): All good. Demos will be published soon.
* SC: I have more general concern about communication. With respect to https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#communication . I was approached weather what I say is my own opinion or it represents the CG findings. I think this is important to make sure that you have consent from the group about what you are communicating. There is a difference between something being presented as one's own opinion or more neutral perspective. Weather that will be capture or not, we still need to be catious about that. This is importatn for chairs, for anybody.

> * TBL: I was thinking that we get a bunch of people and we talk about Solid, I try to understand what failure are you worring about. Is it about distinguishing the output of the group over other outputs. Are you worrying that people will present different uses? Or is it about misrepresenting the wishes of the gruop.

Reviewer 2: ```suggestion
* TBL: I was thinking that we get a bunch of people, and we talk about Solid; I try to understand what failure are you worrying about. Is it about distinguishing the output of the group from other outputs? Are you worrying that people will present different uses? Or is it about misrepresenting the wishes of the group?
```

* SC: It is expressing things that are being worked on by the gruop or something build by individuals or organizations. They can say it is Solid but it might not be worked on in the CG. If the work is presented in the way where it is a subset of the work, or captures aspects of the work which aren't belonging. For example excluding some key parts of the specification but include other parts which have nothing to do with it.

> * HZ: I have similar comment that Tim has, I also want to ask what do you propose?
> * SC: There were some concerns raised about the videos. It is not particular to demos. Mostly how do we communicate externaly.
> * HZ: Communication is never perfect, even PR may not represent the consensus. 

Reviewer 2: ```suggestion
* HZ: I have similar comment that Tim has. I also want to ask, what do you propose?
* SC: There were some concerns raised about the videos. It is not particular to demos. Mostly, how do we communicate externally.
* HZ: Communication is never perfect; even PR may not represent the consensus. 
```

* eP: 
* HZ: Every community requires a level of trust

> * RG: While I'm for having ... communication. Sometimes a person feels strongly about something and they should be able to express it keep it polite.
> * HZ: ..
> * RG: Some discussions have to be done in public. We shouldn't be surpresing it.

Reviewer 2: ```suggestion
* RG: While I'm for having ... communication. Sometimes a person feels strongly about something, and they should be able to express it, but keep it polite.
* HZ: ..
* RG: Some discussions have to be done in public. We shouldn't be surpressing it.
```

* eP: ...
* 

#### TPAC Agenda

* VB (not present, via HackMD): Preparing agenda. Please suggest topics. 
* HZ: I don't know how we can make it actionable

> * TBL: The last one we had it was a bit dysfunctional. Most of the people there had no idea what Solid was, it had to be an introduction. And then we had some discussion among the group participants. We could say that this is not going to be a tutorial. If you want to come, review the website and understand  first. Or we can find out who people are and decide agenda based on that.

Reviewer 2: ```suggestion
* TBL: The last one we had it was a bit dysfunctional. Most of the people there had no idea what Solid was; it had to be an introduction. Then we had some discussion among the group participants. We could say that this is not going to be a tutorial. If you want to come, review the website and understand first. Or, we can find out who people are, and decide agenda based on that.
```

* SC: re "dysfunctional": https://github.com/solid/specification/blob/main/meetings/2023-09-11.md
* SC: I chaired that, not sure about the dysfunctional

> * TBL: Yes it is to strong, it had big problems in trying to be two meetings at once.
> * HZ: I was there, the room was to small and there was some confusion.
> * SC: No one helped me with puting the agenda together. I tried my best given the circumstances, Henry's passing, covering CG charter - which was new and important - WG charter proposal etc.. If the point is if the people should have been prepared, or should we have walked through all the CG reports, possibly. Those were not particularly above and beyond more important than what we covered in the meeting.

Reviewer 2: ```suggestion
* TBL: Yes, it is too strong. It had big problems in trying to be two meetings at once.
* HZ: I was there; the room was too small, and there was some confusion.
* SC: No one helped me with putting the agenda together. I tried my best given the circumstances, Henry's passing, covering CG charter — which was new and important — WG charter proposal, etc. If the point is that people should have been better prepared, or should we have walked through all the CG reports, possibly. Those were not particularly more important than what we covered in the meeting.
```

* TBL: I'm sorry, it was not a personal attack on the chair.

> * RG: Do we have an idea of how many people who will be there know Solid and how many not? We need to know the audience. 
> * HZ: I don't think we may have the idea, we can set our expectations well. 

Reviewer 2: ```suggestion
* RG: Do we have any idea of how many of the people who will be there know Solid, and how many not? We need to know the audience.
* HZ: I don't think we may have much idea. We can set our expectations well.
```

* eP: ...

> * HZ: There is an opportunity to liase with other grups present.

Reviewer 2: ```suggestion
* HZ: There is an opportunity to liaise with other groups present.
```

* SC: We had Hybrid meetings for many years at TPAC. A lot of grups are doing hybrid meetings. For accessibility reason it is important to allow that opportunity. We had meetings for the big part of the year online. If the people want to join and gather in the room they should be able to do that.
* 

### Permalinks

> * eP: The issue of coupling URLs with access control, where initial decision about what the resource is inheriting from can't be changed without changing the URL. We most likely need to document which use cases and requirements the current approach is satisfying, this way we can look at other possible approaches and make sure that they still provide the same behavior. If that's is impossible then we will at least see what specific tradeoffs we need to chose from.
> * RG: One thing that happened, in Solid notifications and in PREP, the access control situation becomes very important. You don't want to compute it every time you are sending notification to each client. With notifications in the picuture it has to be kept in mind.

Reviewer 2: ```suggestion
* eP: There is the issue of coupling URLs with access control, where initial decisions about what the resource is inheriting from can't be changed without changing the URL. We most likely need to document which use cases and requirements are being satisfied by the current approach; this way we can look at other possible approaches and make sure that they still provide the same behavior. If that's impossible, then we will at least see what specific tradeoffs we need to choose from.
* RG: One thing that happened, in Solid notifications and in PREP, the access control situation became very important. You don't want to compute it every time you are sending notification to each client. With notifications in the picture, it has to be kept in mind.
```

> * TBL: I find the corelation between things in the containers and access control useful. The apps that I write use that. It is useful to have access control follow containers you are in. If you don't want to use containers dont't use containers. What I'm trying to find, in your applications if you find access control to be a pain, when people rearange the stuff. Are you imagine that people have things in the containers and that they will move them between containers. The present system works prety well. 
> * HZ: To me it is nothing to do with a container, this is confusion of identity and adresses.

Reviewer 2: ```suggestion
* TBL: I find the correlation between things in the containers and access control useful. The apps that I write use that. It is useful to have access control follow containers you are in. If you don't want to use containers, dont't use containers. What I'm trying to find, if you find access control to be a pain in your applications, when people rearrange the stuff. Are you imagining that people have things in the containers and that they will move them between containers? The present system works prety well.
* HZ: To me, it is nothing to do with a container; this is confusion of identity and adresses.
```

* SC: I understand why it is interesting about moving stuff, that is not what the web architecture says about moving things. By publishing the URI, you commit to it, you may announce persistent policy. If you want to move it, there are thnings like redirects.

> * HZ: In this case you have the service provider in the middle. 
> * eP: When I say move it, I don't want to change the URL but move it do a different access control context.
> * SC: There is a difference with knowledge organization, e.g., putting things into different containers.
> * TBL: Just don't use the containers.

Reviewer 2: ```suggestion
* HZ: In this case, you have the service provider in the middle. 
* eP: When I say move it, I don't want to change the URL, but move it to a different access control context.
* SC: There is a difference in knowledge organization, e.g., putting things into different containers.
* TBL: Just don't use containers.
```

* eP: There are opportunities, for example collaboration with Social CG.
* HZ: This is a culmination of years of work, this validates the work done in the CG. I think that CG stopped progress in last year. Now we should have counteparts in the WG to cooperate with the CG. Thank you everyone, especially PAC, I know this was grinding and stresful effort.
* PAC: Thank you all, this was a group effort.
* RG: Can we have a discussion how are we going to liase with the WG, how do we make contacts etc. 
* PAC: First of all, WG in general work largely in the open and publicly. The mailing list will be open, the github repository will be public. Minutes are published as well. The only thing that is not entirely open are the calls which are reserved to W3C members and invited experts. Also the decisions are made by WG members. Again any member of CG should be able to chime in. WG members can join CG meetings, with their WG hats on. Also WG can invite people for specific meetings. Even if there is this boundry the process is still very, very open.

> * SC: I'm curious what do you mean Hadrian when you say that the progress in the CG stopped in the last year. Which progress do you mean and what do you see as the reason.

Reviewer 3: ```suggestion
* SC: I'm curious about what do you mean Hadrian when you say that the progress in the CG "stopped" in the last year. Which progress do you mean and what do you see as the reason?
```

* HZ: I agree with you, there may be other factors like 
* eP: There is no objective way to measure it, in my subj
* PAC: I don't have an answer, when Sarven asks if the implementation experience will increase. From my experience some organizations are cautious with technologies which are not on the REC track. 
* SC: This is a very likely scenario. As far as the communication, there are things being implemented but it isn't communicated in the CG.
* HZ: This topic was mostly about the announcemtent and celebrating it.
* HZ: As for implementations we have Manas and ActivityPods which were very active in last year. I think there are implemenations on the solid server side.
* eP: We have feedback, for example https://forum.solidproject.org/t/solid-interop-in-practice/7701
* TBL: ... 

### TPAC Demos

* VB (not present, via HackMD): All good. Demos will be published soon.

> * SC: I have more general concern about communication. With respect to https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#communication . I was approached weather what I say is my own opinion or it represents the CG findings. I think this is important to make sure that you have consent from the group about what you are communicating. There is a difference between something being presented as one's own opinion or more neutral perspective. Weather that will be capture or not, we still need to be catious about that. This is importatn for chairs, for anybody.

Reviewer 3: ```suggestion
* SC: I have more general concern about communication with respect to https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#communication . Even when I was chair, I approached that whether what I say out there is my own opinion or it represents the CG's findings. I think this is important to make sure that you have consent from the group about what you are communicating. There is a difference between something being presented as one's own opinion or more neutral perspective. Whether that will be capture or not, we still need to be cautious about that. This is important for chairs and everyone.
```

Reviewer 2: ```suggestion
* SC: I have more general concern about communication with respect to <https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#communication>. Even when I was chair, I approached that whether what I said out there was my own opinion or represented the CG's findings. I think it is important to make sure that you have consent from the group about what you are communicating. There is a difference between something being presented as one's own opinion or a more neutral perspective. Whether that will be captured or not, we still need to be cautious about that. This is important for chairs and everyone.
```

* TBL: I was thinking that we get a bunch of people and we talk about Solid, I try to understand what failure are you worring about. Is it about distinguishing the output of the group over other outputs. Are you worrying that people will present different uses? Or is it about misrepresenting the wishes of the gruop.

> * SC: It is expressing things that are being worked on by the gruop or something build by individuals or organizations. They can say it is Solid but it might not be worked on in the CG. If the work is presented in the way where it is a subset of the work, or captures aspects of the work which aren't belonging. For example excluding some key parts of the specification but include other parts which have nothing to do with it.

Reviewer 3: ```suggestion
* SC: It is expressing things that are being worked on by the group or something built by individuals or organizations. They can say it is Solid but it might not be worked on in the CG. If the work is presented in the way where it is a subset of the work, or captures aspects of the work which aren't belonging. For example excluding some key parts of the specification when communicating but including other parts which have nothing to do with it.
```

Reviewer 2: ```suggestion
* SC: It is about expressing things that are being worked on by the group or something built by individuals or organizations. They can say it is Solid, but it might not be worked on in the CG. If the work is presented in a way where it is a subset of the work, or captures aspects of the work which don't belong. For example, excluding some key parts of the specification when communicating, but including other parts which have nothing to do with it.
```

* SC: re "dysfunctional": https://github.com/solid/specification/blob/main/meetings/2023-09-11.md
* SC: I chaired that, not sure about the dysfunctional
* TBL: Yes it is to strong, it had big problems in trying to be two meetings at once.
* HZ: I was there, the room was to small and there was some confusion.
* SC: No one helped me with puting the agenda together. I tried my best given the circumstances, Henry's passing, covering CG charter - which was new and important - WG charter proposal etc.. If the point is if the people should have been prepared, or should we have walked through all the CG reports, possibly. Those were not particularly above and beyond more important than what we covered in the meeting.
* TBL: I'm sorry, it was not a personal attack on the chair.
* RG: Do we have an idea of how many people who will be there know Solid and how many not? We need to know the audience. 
* HZ: I don't think we may have the idea, we can set our expectations well. 
* eP: ...
* HZ: There is an opportunity to liase with other grups present.

> * SC: We had Hybrid meetings for many years at TPAC. A lot of grups are doing hybrid meetings. For accessibility reason it is important to allow that opportunity. We had meetings for the big part of the year online. If the people want to join and gather in the room they should be able to do that.

Reviewer 3: ```suggestion
* SC: We had Hybrid meetings for some years at TPAC. A lot of groups are doing hybrid meetings. For accessibility reason it is important to allow that opportunity. As eP said, we have meetings for the big part of the year online. If the people want to join and gather in the room they should be able to do that. More the merrier.
```

Reviewer 2: ```suggestion
* SC: We had Hybrid meetings for some years at TPAC. A lot of groups are doing hybrid meetings. For accessibility reasons, it is important to allow that opportunity. As eP said, we have meetings online for most of the year. If people want to join and gather in the room, they should be able to do that. The more the merrier.
```

* RG: One thing that happened, in Solid notifications and in PREP, the access control situation becomes very important. You don't want to compute it every time you are sending notification to each client. With notifications in the picuture it has to be kept in mind.
* TBL: I find the corelation between things in the containers and access control useful. The apps that I write use that. It is useful to have access control follow containers you are in. If you don't want to use containers dont't use containers. What I'm trying to find, in your applications if you find access control to be a pain, when people rearange the stuff. Are you imagine that people have things in the containers and that they will move them between containers. The present system works prety well. 
* HZ: To me it is nothing to do with a container, this is confusion of identity and adresses.

> * SC: I understand why it is interesting about moving stuff, that is not what the web architecture says about moving things. By publishing the URI, you commit to it, you may announce persistent policy. If you want to move it, there are thnings like redirects.

Reviewer 3: ```suggestion
* SC: I understand why it is interesting about moving stuff, but that is not what the Web Architecture says about moving things. When the URI owner allocates a URI to a resource, they commit to it being on the Web. It is a promise when sharing. Like they can be linked to, bookmarked, and they may announce a persistent policy. If you want to move it, there are things like redirects.
```

Reviewer 2: ```suggestion
* SC: I understand why it is interesting around moving stuff, but that is not what Web Architecture says about moving things. When the URI owner allocates a URI to a resource, they commit to it being on the Web. It is a promise when sharing. Like they can be linked to, bookmarked, and they may announce a persistent policy. If you want to move it, there are things like redirects.
```
